### PR TITLE
Partial signed data impl

### PR DESCRIPTION
**Pull request by carl-wallace**
add just enough CMS support to allow incoming path builder implementation to parse certs-only SignedData messages. The inclusion of SetOf<SignerInfo> caused several structs to need Ord and/or PartialOrd added.

### CODE REVIEW

**Comment by tarcieri** [comment]
A general note: the `DerOrd` and `ValueOrd` traits are needed for `SetOf` ordering now. The `Ord` trait does not necessarily reflect the ordering when serialized as DER.

`ValueOrd` can be derived.

File: spki/src/algorithm.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ use der::{
 /// ```
 ///
 /// [RFC 5280 Section 4.1.1.2]: https://tools.ietf.org/html/rfc5280#section-4.1.1.2
-#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
```
Comment (tarcieri) [inline spki/src/algorithm.rs:19]: This type already impls `ValueOrd`

**Review by tarcieri — CHANGES_REQUESTED** [review]

File: spki/src/algorithm.rs (hunk 1 above)
Comment (carl-wallace) [reply spki/src/algorithm.rs]: Dropped Ord and PartialOrd from AlgorithmIdentifier and SignerInfo. Stubbed out ValueOrd for SignerInfo (not needed now and CMS is well short of complete at present).

**Comment by carl-wallace** [comment]
Close pending refactoring to account for recent changes.

**Pull request closed without merging**